packages/plugins/src/plugins/glint-fix.plugin.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  GlintService,
+  Plugin,
+  PluginOptions,
+  PluginsRunnerContext,
+  type PluginResult,
+} from '@rehearsal/service';
+import { applyTextChange, normalizeTextChanges } from '@rehearsal/ts-utils';
+import { CodeFixAction, FileTextChanges, TextChange } from 'typescript';
+import { CodeActionKind, Diagnostic } from 'vscode-languageserver';
+
+export class GlintFixPlugin implements Plugin<PluginOptions> {
+  async run(fileName: string, context: PluginsRunnerContext): PluginResult {
+    const allFixedFiles: Set<string> = new Set();
+    const service = context.service as GlintService;
+
+    const diagnostics = this.getDiagnostics(fileName, service);
+
+    for (const diagnostic of diagnostics) {
```

Comment:
I'm kind of surprised that you don't run into bugs here with looping over the diagnostics and applying changes in the loop. Since we use `diagnostic.range` to compute where we need to splice in the codefix the `range` immediately becomes stale on the next turn of the loop. I think you need to recall `this.getDiagnostics` and re-assign `diagnostics` at the end to ensure you're always getting the new text ranges.

Reply:
Updated this to behave like the `diagnostic-fix` plugin (which does what you suggested)